src/pages/progress/Progress.tsx

```diff
@@ -24,7 +26,7 @@ let particles: any[] = [];
 type Props = {
   changeFlashcardCourseLevel: any;
   flashcardsNextLesson: any;
-  globalUserSettings: any;
+  globalUserSettings: GlobalUserSettings;
```

Comment:
Specified the type in exchange that a couple of `!`s had to be added because type says fields are optional but code uses . I think that's better and not less safer than `any`.

(out of scope, you can ignore) As far as I checked, all fields should exist because it's initialized with non-empty default values ([code](https://github.com/didoesdigital/typey-type/blob/a92a3b6ffde8f209c4bb74f600c0febe49bac92f/src/utils/typey-type.js#L687)). I wonder if there's any reasons if we can remove `?` from the type definition ([code](https://github.com/didoesdigital/typey-type/blob/b55abd93e5ef72c9ea18a153693acdbb5910052e/src/types.ts#L392-L398))

Reply:
Thanks for fixing an `any`!

I believe the `GlobalUserSettings` fields were optional `?` historically because I had deployment challenges I hadn't worked out. For example:

- a user is currently using the site
- a new version of Typey Type is deployed that deletes older JS chunks
- the user then clicks on something on the old version of the site that tries to load a chunk that has now been deleted
- the app explodes

I couldn't be sure if someone in the middle of using the site would have everything they need initialised when they try to load a new chunk of JS.

These days I leave old chunks around.

I *think* if a user loads a JS chunk that expects a new setting to exist (like `backupBannerDismissedTime`) then they would already have loaded the JS that would initialize the setting, so I've removed the type assertions (`!`) and changed the `GlobalUserSettings` types to have required fields.